### ISSUE TITLE
[FEATURE] `script_score` sub-type in rerank processor

### ISSUE THREAD

**Issue by joelthe1** [opensearch-project/neural-search#625]
### Is your feature request related to a problem?
 I have successfully setup and run hybrid search using OpenSearch on my data but I want to tune the results so that the results which are larger in (string) length (for a specific field) are boosted slightly over those results which are relatively smaller in length. The motivation for this is that sometimes, semantic search returns results that are much shorter (think a sentence) than a very long query string (think a paragraph of text) and I would like to reduce the relevance of such results.

### What solution would you like?
Ability to write my own simple script to rerank like in the [script_score](https://opensearch.org/docs/latest/query-dsl/compound/function-score/#the-script-score-function) function.

### What alternatives have you considered?
An inferior alternative would be to use [script_score](https://opensearch.org/docs/latest/query-dsl/compound/function-score/#the-script-score-function) (within function_score ) and compare the length of the query string with the length of the result. But this is comparison is _different_ from the request, since it is a comparison between the individual search result and the query. What I want is to compare all the search results with each other using some simple function (like `script_score`) which is not currently possible.

### Do you have any additional context?
Using the Painless scripting language might be more fitting.

**Comment by navneet1v** [comment]
@joelthe1 if I understand the feature request properly, you are requesting an ability in re-ranking processor which can run a custom logic on the final search results to re-ranking the documents? is that a correct understanding?

**Comment by vamshin** [comment]
@HenryL27  will you be able to take look?

**Comment by HenryL27** [comment]
I'm not gonna have time for this, no. Sorry

**Comment by joelthe1** [comment]
> @joelthe1 if I understand the feature request properly, you are requesting an ability in re-ranking processor which can run a custom logic on the final search results to re-ranking the documents? is that a correct understanding?

Yes @navneet1v, that is correct. Unfortunately, I am not in a position to work on this myself but do think it would benefit more than just me.

**Comment by heemin32** [comment]
@joelthe1 Could you provide more details to clarify the request? For example, an API example and the expected behavior would be helpful.
